src/main/java/io/split/android/client/storage/attributes/AttributesStorageImpl.java

```diff
@@ -0,0 +1,80 @@
+package io.split.android.client.storage.attributes;
+
+import androidx.annotation.Nullable;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AttributesStorageImpl implements AttributesStorage {
+
+    @Nullable private final PersistentAttributesStorage mPersistentAttributesStorage;
+    private final Map<String, Object> mInMemoryAttributes = new ConcurrentHashMap<>();
+
+    public AttributesStorageImpl(@Nullable PersistentAttributesStorage persistentAttributesStorage) {
+        mPersistentAttributesStorage = persistentAttributesStorage;
+    }
+
+    @Override
+    public void loadLocal() {
+        if (mPersistentAttributesStorage != null) {
+            overwriteValuesInMemory(mPersistentAttributesStorage.getAll());
+        }
+    }
+
+    @Nullable
+    @Override
+    public Object get(String key) {
+        return mInMemoryAttributes.get(key);
+    }
+
+    @Override
+    public Map<String, Object> getAll() {
+        return mInMemoryAttributes;
```

Comment:
This function should return a copy (not concurrent) of mInMemoryAttributes map to avoid accidentally modifying the original one.